app/src/main/java/com/example/atheneum/activities/AddEditBookActivity.java

```diff
@@ -92,6 +98,26 @@ protected void onCreate(Bundle savedInstanceState) {
         isbnEditText = findViewById(R.id.isbnEditText);
         descEditText = findViewById(R.id.descEditText);
 
+        final String bookID = getIntent().getStringExtra("BookID");
+        if( bookID != ""){
+            Log.i("got intent extra bookid", bookID);
+
+            BookInfoViewModelFactory factory = new BookInfoViewModelFactory(bookID);
+            bookInfoViewModel = ViewModelProviders.of(this, factory).get(BookInfoViewModel.class);
+            LiveData<Book> bookLiveData = bookInfoViewModel.getBookLiveData();
+            bookLiveData.observe(this, new Observer<Book>() {
+                @Override
+                public void onChanged(@Nullable Book book) {
+                    if (book != null) {
+                        titleEditText.setText(book.getTitle());
+                        authorEditText.setText(book.getAuthor());
+                        isbnEditText.setText(String.valueOf(book.getIsbn()));
+                        descEditText.setText(book.getDescription());
+                    }
```

Comment:
If you're intending on auto-filling the book information, you should call `removeObserver(this)` at the end of onChanged in order to prevent the UI updating based on data changes. Although that shouldn't happen in this case, it's still a good idea to remove the observer if you only want the data on startup and don't want to continuously observe the data.

Reply:
Ok, its added